inc/targetticket.class.php

```diff
@@ -1095,218 +1101,246 @@ public function save(PluginFormcreatorFormAnswer $formanswer) {
       ];
 
       $data = Ticket::getDefaultValues();
-      $ticket  = new Ticket();
       $form    = $formanswer->getForm();
       $answer  = new PluginFormcreatorAnswer();
 
-      $data['requesttypes_id'] = PluginFormcreatorCommon::getFormcreatorRequestTypeId();
-
-      // Get predefined Fields
-      $ttp                  = new TicketTemplatePredefinedField();
-      $predefined_fields    = $ttp->getPredefinedFields($this->fields['tickettemplates_id'], true);
-
-      if (isset($predefined_fields['_users_id_requester'])) {
-         $this->addActor('requester', $predefined_fields['_users_id_requester'], true);
-         unset($predefined_fields['_users_id_requester']);
-      }
-      if (isset($predefined_fields['_users_id_observer'])) {
-         $this->addActor('observer', $predefined_fields['_users_id_observer'], true);
-         unset($predefined_fields['_users_id_observer']);
+      $createTicket = false;
+      if(!empty($this->fields['has_condition'])) {
+         $conditionValues = $DB->request([
+            'SELECT' => ['answer'],
+            'FROM'   => PluginFormcreatorAnswer::getTable(),
+            'WHERE'  => [
+               'plugin_formcreator_formanswers_id' => $formanswer->fields['id'],
+               'plugin_formcreator_questions_id'   => $this->fields['condition_question']
+            ]
+         ])->next();
+         $conditionValues = $conditionValues['answer'];
+         $conditionValuesArray = preg_split("/\\r\\n/", $conditionValues);
+         foreach ($conditionValuesArray as $conditionValue) {
+            if($conditionValue == $this->fields['condition_value']) {
+               $createTicket = true;
+            }
+         }
+         
       }
-      if (isset($predefined_fields['_users_id_assign'])) {
-         $this->addActor('assigned', $predefined_fields['_users_id_assign'], true);
-         unset($predefined_fields['_users_id_assign']);
+      else {
+         $createTicket = true;
       }
 
-      if (isset($predefined_fields['_groups_id_requester'])) {
-         $this->addGroupActor('assigned', $predefined_fields['_groups_id_requester']);
-         unset($predefined_fields['_groups_id_requester']);
-      }
-      if (isset($predefined_fields['_groups_id_observer'])) {
-         $this->addGroupActor('observer', $predefined_fields['_groups_id_observer']);
-         unset($predefined_fields['_groups_id_observer']);
-      }
-      if (isset($predefined_fields['_groups_id_assign'])) {
-         $this->addGroupActor('assigned', $predefined_fields['_groups_id_assign']);
-         unset($predefined_fields['_groups_id_assign']);
-      }
+      if($createTicket === true) {
```

Comment:
If you reverse this condition then tou will not need to change a lot of code. This will make the diff of the PR easier to review, and the final code easier to read.